### PR TITLE
refactor!: index node metadata consistency and improvements

### PR DESCRIPTION
**Pull request by janaka**
## Description

Help PR #66 implement a cleaner solution.

- fix: metadata key names back to the `DocumentMetadata` enum key name in lowercase
- refactor!: set `data_source_type` to the data sources' respective base class name.
- add: `data_source_name` which is set from the data source name property.
- add: missing metadata to Azure Blob data source
- fix: incorrect type for `last_modifed` value for |zure document list. (convert to epoch)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactor and code improvement (non-breaking change which improves code quality and/or performance)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [x] Tests
- [ ] Other chores such as maintenance

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration by modifying the list below.

- [ ] Test A
- [ ] Test B
- [ ] Test C

**Test Configuration**:

Please describe the test setup. List them below as bullet points.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings
- [ ] The commit message follows the convention of this project

### CODE REVIEW

**Comment by janaka** [comment]
note: I'm trying to add some unit tests for the data source metadata. shouldn't alter the I'm implementation.

**Comment by janaka** [comment]
add a test for one data source to check expected meta data is added to to Documents. We can increase coverage on a separate PR.

File: tests/docq/data_source/manual_upload_test.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+            assert document_list == [
+                (
+                    ".persisted/upload/PERSONAL/1000/www_gartner_com_doc_reprints_id_1_2BZFRDHS_ct_221212_st_sb.pdf",
+                    1234567890,
+                    1024,
+                )
+            ]
+
+    def test_load(self) -> None:
```
Comment (janaka) [inline tests/docq/data_source/manual_upload_test.py:47]: This is more of an integration test but seems like the best way to test metadata without getting into deep mocking/patching. We can use a smaller PDF to reduce execution time if needed.

File: source/docq/data_source/azure_blob.py [hunk 2]
```diff
@@ -41,6 +42,19 @@ def get_config_keys(self) -> List[ConfigKey]:
...
+            doc.metadata.update(
+                {
+                    str(DocumentMetadata.SPACE_ID.name).lower(): space.id_,
+                    str(DocumentMetadata.SPACE_TYPE.name).lower(): space.type_.name,
+                    str(DocumentMetadata.DATA_SOURCE_NAME.name).lower(): self.get_name(),
+                    str(DocumentMetadata.DATA_SOURCE_TYPE.name).lower(): self.__class__.__base__.__name__,
+                }
+            )
+            return doc
+
```
Comment (osala-eng) [inline source/docq/data_source/azure_blob.py:57]: Still no metadata returned for Azure Blob

**Review by osala-eng — COMMENTED** [review]
I re-indexed and tested, Both `Web Scrapper` and `Manual Upload` work as expected and I still get empty metadata for `Azure Blob` and `AWS S3`

**Review by osala-eng — APPROVED** [review]
LGTM

File: source/docq/data_source/azure_blob.py [hunk 3]
```diff
@@ -41,6 +42,19 @@ def get_config_keys(self) -> List[ConfigKey]:
...
         """Load the documents from azure blob container."""
+
+        # str(DocumentMetadata.DATA_SOURCE_TYPE.name).lower(): self.__class__.__base_.__name__,
+        def lambda_apend_metadata(doc: Document) -> Document:
+            doc.metadata.update(
+                {
+                    str(DocumentMetadata.SPACE_ID.name).lower(): space.id_,
+                    str(DocumentMetadata.SPACE_TYPE.name).lower(): space.type_.name,
+                    str(DocumentMetadata.DATA_SOURCE_NAME.name).lower(): self.get_name(),
+                    str(DocumentMetadata.DATA_SOURCE_TYPE.name).lower(): self.__class__.__base__.__name__,
```
Comment (cwang) [inline source/docq/data_source/azure_blob.py:53]: Not sure why we needed to use lowercase - any case works as a key?

File: tests/docq/data_source/manual_upload_test.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            )
+            self.assertAlmostEqual(  # noqa: PT009
+                documents[0].metadata[str(DocumentMetadata.INDEXED_ON.name).lower()],
+                datetime.timestamp(datetime.now().utcnow()),
+                delta=5,
+            )
+
+            # SimpleDirectoryReader generated meta data fields that we depend on.
+            assert documents[0].metadata["file_name"] == "Research-Revealing-the-True-GenAI-Data-Exposure-Risk.pdf"
+            assert documents[0].metadata["page_label"] is not None
```
Comment (cwang) [inline tests/docq/data_source/manual_upload_test.py:80]: Nice 👍

File: source/docq/data_source/main.py [hunk 5]
```diff
@@ -16,6 +16,7 @@ class DocumentMetadata(Enum):
     SPACE_ID = "Space ID"
     SPACE_TYPE = "Space Type"
     DATA_SOURCE_TYPE = "Data Source Type"
+    DATA_SOURCE_NAME = "Data Source Name"
```
Comment (cwang) [inline source/docq/data_source/main.py:19]: I'm guessing this is upon request from @osala-eng ? I thought an alternative is to use space id and space type to get to the data source type but it's fine if this is easier.

**Review by cwang — COMMENTED** [review]
A few comments

**Pull request merged**